### DISCUSSION TITLE
shared state that's accessible for the webserver?

### DISCUSSION THREAD

**Discussion by Zerowalker** [dotnet/WatsonWebserver#115] [Q&A] [answered]
How do you make a shared state that's accessible within a route?

Currently i just make it static and global, but i would rather have it embedded in the context or something.

Thanks:)

**Comment by jchristn** [comment, marked as answer]
Hi, there are a couple of immediate options.

1) add a pre-request handler that will attach your state to the ```HttpContext.Metadata (object)``` property.

2) use a static class that is ```public```/```internal``` that the assembly/routes can reach.

Cheers, Joel

**Reply by Zerowalker** [reply]
Ah got it, will, thanks!:)
